security/patch-per-service-tls.yml

```diff
@@ -0,0 +1,247 @@
+          - id: truststore
+            type: secret
+            mountPath: /truststore
+      echo:
+        # Mount the keystore and truststore secrets to the container
+        kubernetes:
+          volumes:
+          - id: echo-cert
+            type: secret
+            mountPath: /echo-cert
+          - id: truststore
+            type: secret
+            mountPath: /truststore
+      fiat:
+        # Mount the keystore and truststore secrets to the container
+        kubernetes:
+          volumes:
+          - id: fiat-cert
+            type: secret
+            mountPath: /fiat-cert
+          - id: truststore
+            type: secret
+            mountPath: /truststore
+      gate:
+        # Mount the keystore and truststore secrets to the container
+        kubernetes:
+          volumes:
+          - id: gate-cert
+            type: secret
+            mountPath: /gate-cert
+          - id: truststore
+            type: secret
+            mountPath: /truststore
+      igor:
+        # Mount the keystore and truststore secrets to the container
+        kubernetes:
+          volumes:
+          - id: igor-cert
+            type: secret
+            mountPath: /igor-cert
+          - id: truststore
+            type: secret
+            mountPath: /truststore
+      kayenta:
+        # Mount the keystore and truststore secrets to the container
+        kubernetes:
+          volumes:
+          - id: kayenta-cert
+            type: secret
+            mountPath: /kayenta-cert
+          - id: truststore
+            type: secret
+            mountPath: /truststore
+      rosco:
+        # Mount the keystore and truststore secrets to the container
+        kubernetes:
+          volumes:
+          - id: rosco-cert
+            type: secret
+            mountPath: /rosco-cert
+          - id: truststore
+            type: secret
+            mountPath: /truststore
+      dinghy:
+        # Mount the certs to the container.  The Truststore is not needed as this is a GoLang Service.
+        kubernetes:
+          volumes:
+          - id: dinghy-cert
+            type: secret
+            mountPath: /dinghy-cert
+      terraformer:
+        # Mount the certs to the container.  The Truststore is not needed as this is a GoLang Service.
+        kubernetes:
+          volumes:
+          - id: terraformer-cert
+            type: secret
+            mountPath: /terraformer-cert
+    profiles:
+      front50:
+        server:
+          ssl:
+            # Enable TLS and set the keystore, certs, and trustore for the Server component of the service
+            enabled: true
+            key-store: /front50-cert/keystore.jks
+            key-store-type: jks
+            key-store-password: changeit # This should match the password set when the keystore was created
+            key-alias: front50 # This should match the alias used when adding the cert to the keystore
+            trust-store: /truststore/truststore.jks
```

Comment:
I'm wondering if mounting volumes like this is needed, can you try using `encryptedFile:k8s` secret references instead? The kubernetes secret engine already mounts files coming from kubernets secrets into the service pods, that could save a lot of trouble setting up individual volumes.

If using the kubernetes secret engine doesn't work, I'll suggest that you also add in `kustomization-all.yml` the configuration needed to actually create the secrets from files. There's already an example [here](https://github.com/armory/spinnaker-kustomize-patches/blob/master/recipes/kustomization-all.yml#L85)

Reply:
I'm not as sure if this makes sense. Just need the supporting doc on creating and mounting these truststore and keystore files. For now I'm ok with this approach, and maybe another future task is to look to see if using secret references would work. Either way.

Reply:
I'm in favor of having full working configs as much as possible, and if we can avoid going to a separate doc for making this work the better. If not, I think the doc should at least be linked here as a comment.

Reply:
Does the encryptedFile thing work with these values (specifically the keystore, truststore, cert, ca, and key values)?  I know this "volumes" method works.  I've had trouble recently with support in some fields for the k8s secrets.  I'll give it a try when I have time, but it might have to be a change down the road.

Reply:
I don't know if those fields will work. It's you call to try it or link to the doc that Michael mentioned and/or add the example to `kustomization-all.yml` for how to create the secret with the keyStore/truststore files.

Reply:
There's no doc yet.  I still have to write it.  Also, the keystores, certs, etc are created by cert-manager.  The truststore has to be created manually and loaded into a secret.